src/ExampleApp.cpp

```diff
@@ -117,143 +132,316 @@ ExampleApp::ExampleApp(Eegeo::EegeoWorld* pWorld,
-	m_pExampleController->RegisterCameraExample<Examples::ToggleTrafficExampleFactory>();
-	m_pExampleController->RegisterCameraExample<Examples::TrafficCongestionExampleFactory>();
-	m_pExampleController->RegisterCameraExample<Examples::WebRequestExampleFactory>();
+    m_pExampleController->RegisterCameraExample<Examples::ToggleTrafficExampleFactory>();
+    m_pExampleController->RegisterCameraExample<Examples::TrafficCongestionExampleFactory>();
+    m_pExampleController->RegisterCameraExample<Examples::WebRequestExampleFactory>();
     m_pExampleController->RegisterCameraControllerScreenPropertiesProviderExample<Examples::RenderToTextureExampleFactory>(m_screenPropertiesProvider);
+	#ifdef CARDBOARD
+    m_pExampleController->RegisterScreenPropertiesProviderVRExample<Examples::VRCardboardExampleFactory>(m_screenPropertiesProvider);
+	#endif
 }
 
 ExampleApp::~ExampleApp()
 {
-	delete m_pCameraTouchController;
+    delete m_pCameraTouchController;
     delete m_pLoadingScreen;
     delete m_pExampleController;
+	#ifdef CARDBOARD
+    Eegeo_DELETE m_pVRSkybox;
+    Eegeo_DELETE m_pVRDistortion;
+
+    Eegeo_DELETE m_pStreamingVolume;
+	#endif
 }
 
 void ExampleApp::OnPause()
 {
-	Eegeo::EegeoWorld& eegeoWorld = World();
-	eegeoWorld.OnPause();
+    Eegeo::EegeoWorld& eegeoWorld = World();
+    eegeoWorld.OnPause();
 }
 
 void ExampleApp::OnResume()
 {
-	Eegeo::EegeoWorld& eegeoWorld = World();
-	eegeoWorld.OnResume();
+    Eegeo::EegeoWorld& eegeoWorld = World();
+    eegeoWorld.OnResume();
 }
 
-void ExampleApp::Update (float dt)
+void ExampleApp::Update (float dt, float headTansform[])
 {
-	Eegeo::EegeoWorld& eegeoWorld = World();
-    
+    Eegeo::EegeoWorld& eegeoWorld(World());
+    eegeoWorld.EarlyUpdate(dt);
+    Eegeo::Camera::CameraState cameraState(m_pExampleController->GetCurrentCameraState());
+
     m_pCameraTouchController->Update(dt);
 
-	eegeoWorld.EarlyUpdate(dt);
-    
-	m_pExampleController->EarlyUpdate(dt);
-    
-    Eegeo::Camera::CameraState cameraState(m_pExampleController->GetCurrentCameraState());
-    Eegeo::Streaming::IStreamingVolume& streamingVolume(m_pExampleController->GetCurrentStreamingVolume());
-    
-    Eegeo::EegeoUpdateParameters updateParameters(dt,
-                                                  cameraState.LocationEcef(),
-                                                  cameraState.InterestPointEcef(),
-                                                  cameraState.ViewMatrix(),
-                                                  cameraState.ProjectionMatrix(),
-                                                  streamingVolume,
-                                                  m_screenPropertiesProvider.GetScreenProperties());
-    
-	eegeoWorld.Update(updateParameters);
-    
-    m_pExampleController->Update(dt);
-    
+    m_pExampleController->EarlyUpdate(dt);
+
+    if(m_pExampleController->GetCurrentExampleName() == "VRCardboardExample")
```

Comment:
This looks a bit iffy. We're breaking the ExampleController abstraction by having to know what example is running. Ideally we shouldn't need to know which example is running.